### ISSUE TITLE
Spatial Soft-argmax (+optional scale adaptive)

### ISSUE THREAD

**Issue by bhack** [tensorflow/addons#1364]
**Describe the feature and the current behavior/state.**
Recover Spatial Soft-argmax from contrib + scale adaptive extension

**Relevant information**
- Are you willing to contribute it (yes/no): no
- Are you willing to maintain it going forward? (yes/no): yes
- Is there a relevant academic paper? (if so, where): https://arxiv.org/abs/2003.07543 (for the scale adaptive version)
- Is there already an implementation in another framework? (if so, where): [Kornia](https://github.com/kornia/kornia/blob/master/kornia/geometry/spatial_soft_argmax.py#L483) (Scale adaptation extended formulation is still not released)
- Was it part of tf.contrib? (if so, where): https://github.com/tensorflow/tensorflow/issues/23331 (only the non-scale adaptive version)

**Which API type would this fall under (layer, metric, optimizer, etc.)**
`layer`

**Who will benefit with this feature?**
Any application that requires heatmap based keypoints/landmarks regression

**Any other info.**

**Comment by shreyashpatodia** [comment]
I can take a look at this in the weekend if you guys still need help?

**Comment by bhack** [comment]
@shreyashpatodia How do you want to organize the scale parameter API? With the scale adaptable extension the spatial soft argmax ops need to work inside ROI windows in every channel.

**Comment by bhack** [comment]
It partially relate on how we want to design basic keypoints<->heatmap ops. See https://github.com/tensorflow/addons/issues/1366

**Comment by bhack** [comment]
I think that something like [`tf.image.crop_to_bounding_box`](https://www.tensorflow.org/api_docs/python/tf/image/crop_to_bounding_box) but that works with a batch of offsets it is needed as parameter to work with Scales.

**Comment by bhack** [comment]
Probably the ROIs could be formulated like `boxes` in [`tf.image.crop_to_bounding_box`](https://www.tensorflow.org/api_docs/python/tf/image/draw_bounding_boxes?version=nightly)

**Comment by bhack** [comment]
@shreyashpatodia Do you plan to make a PR?

**Comment by bhack** [comment]
@gabrieldemarmiesse @seanpmorgan Do you remember why we lost the `tf.contrib` version in the migration? In the [table](https://github.com/tensorflow/community/blob/master/rfcs/20181214-move-to-addons.md) I see:
> One OSS project found / Needs refactored as base Layer subclass / Uses get_variable_collections

**Comment by seanpmorgan** [comment]
> @gabrieldemarmiesse @seanpmorgan Do you remember why we lost the `tf.contrib` version in the migration? In the [table](https://github.com/tensorflow/community/blob/master/rfcs/20181214-move-to-addons.md) I see:
> 
> > One OSS project found / Needs refactored as base Layer subclass / Uses get_variable_collections

Just means there wasn't enough OSS usage to justify moving it. During the RFC period all candidates to be dropped were welcome to be picked up if there was community support, but no one did for `spatial_softmax`. Happy to review a PR for it, but the tf.contrib version used variable scoping etc. so there are some changes to be made.

**Comment by bhack** [comment]
What do you think about the Google version that I mentioned early in this ticket?

**Comment by bhack** [comment]
I was looking at `pytorch.vision` [API signature for max pooling](https://pytorch.org/docs/stable/torchvision/ops.html#torchvision.ops.roi_pool) and I think that we could have a quite similar API interface for roi spatial soft-max.

**Comment by bhack** [comment]
Also `boxes` and `box_indices` params design in [crop_and_resize] Tensorflow API (https://www.tensorflow.org/api_docs/python/tf/image/crop_and_resize) could be ok.

**Comment by bhack** [comment]
I would close this tomorrow but if somone  is still interested in this topic I will just unsubscribe to the ticket. Let me know.

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
